src/windows/scanner.cc

```diff
@@ -220,15 +220,23 @@ ScanMountpoints(drivelist::com::Connection *const connection,
     if (FAILED(result))
       return InterpretHRESULT(result);
 
-    BOOL isProtected;
-    result = drivelist::volume::GetReadOnlyFlag(path[0], &isProtected);
+    // Turns out a disk can be writable while its volumes can be read-only
+    // and viceversa, so we need to check the writable capabilities of
+    // both the disk and its volumes
+    BOOL writable;
+    result = drivelist::volume::IsDiskWritable(path[0], &writable);
     if (FAILED(result))
       return InterpretHRESULT(result);
+    if (writable && hasFilesystem) {
+      result = drivelist::volume::IsVolumeWritable(path[0], &writable);
+      if (FAILED(result))
+        return InterpretHRESULT(result);
+    }
 
     drivelist::mountpoint_s mountpoint;
     mountpoint.path = path;
     mountpoint.disk = "\\\\.\\PHYSICALDRIVE" + std::to_string(number);
-    mountpoint.readonly = isProtected;
+    mountpoint.readonly = !writable;
     mountpoint.system = path[0] == systemDriveLetter;
     mountpoint.hasFilesystem = hasFilesystem;
```

Comment:
I mean, we expose the read-only drive, but not the ones without a file-system, even though we still want to use some of its metadata for the top level disk.

For example, in Windows, we only determine if a top level drive is read-only if one of its volumes is read-only. There are drives that have no file-system, and even though Windows assigns it a letter in the background, is not something user can even see from the file explorer. By adding a boolean flag here, we can then check if its read-only or not, and set the top level property accordingly, but still filter it out from the final set of volumes.